cps_stage3/extrapolation.py

```diff
@@ -102,19 +102,19 @@ def _extrapolate(WT, I, benefits, prob, target, tol=0.01, J=15):
         diff = actual - target
         if diff < 0:
             remove = False
-            candidates = extrap_df.loc[extrap_df.I == 0, ]
+            candidates = extrap_df.loc[extrap_df.I == 0, ].copy()
             assert candidates.I.sum() == 0
             # set everyone as a participant and then remove surplus below
             # this is necessary since I * wt will all be zeroes
             candidates.loc[:, "I"] = np.ones(len(candidates))
             candidates.loc[:, "I_wt"] = candidates.I * candidates.wt
-            noncandidates = extrap_df.loc[extrap_df.I == 1, ]
+            noncandidates = extrap_df.loc[extrap_df.I == 1, ].copy()
             assert noncandidates.I.sum() == len(noncandidates)
         else:
             remove = True
-            candidates = extrap_df.loc[extrap_df.I == 1, ]
+            candidates = extrap_df.loc[extrap_df.I == 1, ].copy()
             assert candidates.I.sum() == len(candidates)
-            noncandidates = extrap_df.loc[extrap_df.I == 0, ]
+            noncandidates = extrap_df.loc[extrap_df.I == 0, ].copy()
```

Comment:
@martinholmer For some reason, I had to use the `copy()` method here. I thought that you were safe if you used the `loc` accessor method instead of doing something like `candidates = extrap_df[extrap_df.I == 1]`. 

Does any one have any thoughts on why I had to use the copy method here?